### PR TITLE
[action] Add extra_arg parameter to oclint action

### PR DESCRIPTION
**Pull request by mourad-brahim**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by janpio** [comment]
You will have to add this new parameter at the end of the options list as well: https://github.com/fastlane/fastlane/blob/3630648c264551d40dea1fc3df2f5806bab19c02/fastlane/lib/fastlane/actions/oclint.rb#L205-L210

File: fastlane/lib/fastlane/actions/oclint.rb [hunk 1]
```diff
@@ -207,6 +207,11 @@ def self.available_options
                                        description: "Allow duplicated violations in the OCLint report",
                                        is_string: false,
                                        default_value: false)
```
Comment (janpio) [inline fastlane/lib/fastlane/actions/oclint.rb]: this needs a comma at the end

File: fastlane/lib/fastlane/actions/oclint.rb [hunk 2]
```diff
@@ -207,6 +207,11 @@ def self.available_options
...
+          FastlaneCore::ConfigItem.new(key: :extra_arg,
+                                       env_name: 'FL_OCLINT_EXTRA_ARG',
+                                       description: 'Additional argument to append to the compiler command line',
+                                       default_value: '',
+                                       optional: true),
```
Comment (janpio) [inline fastlane/lib/fastlane/actions/oclint.rb]: and this doesn't need a comma at the end

**Comment by janpio** [comment]
```
fastlane/lib/fastlane/actions/oclint.rb:83:1: C: Layout/TrailingWhitespace: Trailing whitespace detected.
fastlane/lib/fastlane/actions/oclint.rb:85:1: C: Layout/TrailingWhitespace: Trailing whitespace detected.
```

File: fastlane/lib/fastlane/actions/oclint.rb [hunk 3]
```diff
@@ -204,7 +206,12 @@ def self.available_options
                                        env_name: "FL_OCLINT_ALLOW_DUPLICATED_VIOLATIONS",
                                        description: "Allow duplicated violations in the OCLint report",
                                        is_string: false,
-                                       default_value: false)
+                                       default_value: false),
+          FastlaneCore::ConfigItem.new(key: :extra_arg,
+                                       env_name: 'FL_OCLINT_EXTRA_ARG',
+                                       description: 'Additional argument to append to the compiler command line',
+                                       default_value: '',
```
Comment (janpio) [inline fastlane/lib/fastlane/actions/oclint.rb]: This probably should not have a `default_value` if it is optional, correct?

Comment (mourad-brahim) [reply fastlane/lib/fastlane/actions/oclint.rb]: Correct :)

**Comment by janpio** [comment]
That looks good @mourad-brahim. 
Do you maybe want to also take a look at the tests and add test cases for when this option is used?

**Review by joshdholtz — APPROVED** [review]
This looks good! Thanks for the contribution and thanks to @janpio for the review ❤️

**Pull request merged**